client/src/components/App.jsx

```diff
@@ -21,8 +21,8 @@ export default class App extends React.Component {
     this.setRelatedProducts(productId);
   }
 
-  componentWillReceiveProps() {
-    this.setRelatedProducts(this.props.id);
+  componentWillReceiveProps(nextProps) {
```

Comment:
@MichaelDimitras I saw that getDerivedStateFromProps was a static method that didn't have access to "this" so I didn't think I could, but I see now that if you return something from that method it will set the state -- so I will merge this bug fix, then submit another PR to do things the react 16 way. Good catch!